File: libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/navigation/NavigationRouteEventListenerTest.java
package com.mapbox.services.android.navigation.v5.navigation;

import org.junit.Test;

import okhttp3.Call;

import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.verify;

public class NavigationRouteEventListenerTest {

  @Test
  public void callStart_timeStartIsCalled() {
    ElapsedTime time = mock(ElapsedTime.class);
    NavigationRouteEventListener listener = new NavigationRouteEventListener(time);

    listener.callStart(mock(Call.class));

    verify(time).start();
  }

  @Test
  public void callEnd_timeEndIsCalled() {
    ElapsedTime time = mock(ElapsedTime.class);
    NavigationRouteEventListener listener = new NavigationRouteEventListener(time);

    listener.callEnd(mock(Call.class));

    verify(time).end();
  }
}

File: libdirections-offboard/src/test/java/com/mapbox/navigation/route/offboard/router/NavigationRouteEventListenerTest.kt
package com.mapbox.navigation.route.offboard.router

import com.mapbox.navigation.utils.time.ElapsedTime
import io.mockk.mockk
import io.mockk.verify
import okhttp3.Call
import org.junit.Test

class NavigationRouteEventListenerTest {

    @Test
    fun callStart_timeStartIsCalled() {
        val time = mockk<ElapsedTime>(relaxed = true)
        val call = mockk<Call>(relaxed = true)
        val listener = NavigationRouteEventListener(time)

        listener.callStart(call)

        verify { time.start() }
    }

    @Test
    fun callEnd_timeEndIsCalled() {
        val time = mockk<ElapsedTime>(relaxed = true)
        val call = mockk<Call>(relaxed = true)
        val listener = NavigationRouteEventListener(time)

        listener.callEnd(call)

        verify { time.end() }
    }
}
